### DISCUSSION TITLE
Welcome to FastNoise2 Discussions!

### DISCUSSION THREAD

**Discussion by Auburn** [Auburn/FastNoise2#132] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Aurailus** [comment]
Hi! I'm Auri, and I'm using FastNoise2 in my WIP voxel engine. I swapped over from libnoise when I wasn't getting the performance I wanted, and I've been blown away by how well it's performed in comparison. It's allowed me to get a lot more crazy with my world generation and render distance because of the extra speed. It's a really cool project, and I'm super excited to see how it develops over time. 

Something that I've been anxiously awaiting is the ability to get the grade of a point/region from noise modules in addition to the values of those points. I remember seeing an issue where you said it was planned, and it would be hugely helpful to me. I would try to contribute myself, but all this SIMD stuff is way beyond me.

I'm also curious if you're planning on writing more documentation about the functionality of each Noise Module? I don't know every single detail about the library, but I'd be happy to contribute what I can to that if it would be helpful. I found there was a big learning curve for me when first figuring out what was available is FastNoise2 because there weren't any docs.

**Reply by Auburn** [reply]
Thanks for the praise, I realise the documentation is lacking it's not a strong point of mine 😅 As part of each node type's metadata there is a description of the node's features and similar description for each variable. I've not filled this in for every node type yet but there is more progress on the newFastSIMD branch. In the Noise Tool these descriptions can be seen by hovering on the node title or variable names. Given all the data is there it would be good to make a tool that extracts it into a markdown format and maybe generate a preview texture for each node type, the markdown could then be uploaded to GitHub documentation.

The gradient/derivative output for the noise generation is partly a question of how it integrates into the structure of the library. I have done some initial work on this and think I have a reasonable solution. But the current priority is to get v1.0 (newFastSIMD) finalised.

Any help would be appreciated, I'm happy to help guide through the SIMD stuff, it's mostly abstracted but I realise it's still daunting. Implementing the derivate calculations for each noise algorithm is probably the most time consuming part. If you do want to look into this further it's probably easier to join the discord linked in the readme for easier back and forth.
